File: Serene/Serene.Script/Imports/ServiceContracts/Administration.UserRoleService.cs
namespace Serene.Administration
{
    using jQueryApi;
    using Serenity;
    using System;
    using System.Collections;
    using System.Collections.Generic;
    using System.Runtime.CompilerServices;

    [Imported, PreserveMemberCase]
    public partial class UserRoleService
    {
        [InlineConstant] public const string BaseUrl = "Administration/UserRole";
    
        [InlineCode("Q.serviceRequest('Administration/UserRole/Update', {request}, {onSuccess}, {options})")]
        public static jQueryXmlHttpRequest Update(UserRoleUpdateRequest request, Action<SaveResponse> onSuccess, ServiceCallOptions options = null)
        {
            return null;
        }
    
        [InlineCode("Q.serviceRequest('Administration/UserRole/List', {request}, {onSuccess}, {options})")]
        public static jQueryXmlHttpRequest List(UserRoleListRequest request, Action<UserRoleListResponse> onSuccess, ServiceCallOptions options = null)
        {
            return null;
        }
    
        [Imported, PreserveMemberCase]
        public static class Methods
        {
            [InlineConstant] public const string Update = "Administration/UserRole/Update";
            [InlineConstant] public const string List = "Administration/UserRole/List";
        }
    }
    
}

File: Serene/Serene.Web/Modules/Imports/ServerTypings/Administration.UserRoleService.ts
namespace Serene.Administration {
    export namespace UserRoleService {
        export const baseUrl = 'Administration/UserRole';

        export declare function Update(request: UserRoleUpdateRequest, onSuccess?: (response: Serenity.SaveResponse) => void, opt?: Serenity.ServiceOptions<any>): JQueryXHR;
        export declare function List(request: UserRoleListRequest, onSuccess?: (response: UserRoleListResponse) => void, opt?: Serenity.ServiceOptions<any>): JQueryXHR;

        export namespace Methods {
            export declare const Update: 'Administration/UserRole/Update';
            export declare const List: 'Administration/UserRole/List';
        }

        ['Update', 'List'].forEach(x => {
            (<any>UserRoleService)[x] = function (r, s, o) { return Q.serviceRequest(baseUrl + '/' + x, r, s, o); };
            (<any>Methods)[x] = baseUrl + '/' + x;
        });
    }
}

